tutorials/time-series-visualization-with-altair/tutorial.ipynb

```diff
@@ -0,0 +1,2133 @@
+{
```

Comment:
<p>I think that setting global default args that are overwritten as you proceed through the notebook may be confusing to readers and error-prone, and is not an approach that we want to highlight. I think a better approach would be to create a function that returns functions that can be mapped over image collections. For example:</p><p> </p><p> </p><p><code>def create_reduce_region_function(geometry,</code></p><p><code>                                  reducer=ee.Reducer.mean(),</code></p><p><code>                                  crs='EPSG:4326',</code></p><p><code>                                  scale=1000,</code></p><p><code>                                  bestEffort=True,</code></p><p><code>                                  maxPixels=1e14,</code></p><p><code>                                  tileScale=4</code></p><p><code>                                  ): </code></p><p><code>    def reduce_region_function(img):</code></p><p><code>      stat = img.reduceRegion(</code></p><p><code>        reducer=reducer,</code></p><p><code>        geometry=geometry,</code></p><p><code>        scale=scale,</code></p><p><code>        crs=crs,</code></p><p><code>        bestEffort=bestEffort,</code></p><p><code>        maxPixels=maxPixels,</code></p><p><code>        tileScale=tileScale)</code></p><p><code>      return ee.Feature(geometry, stat).set({'millis': img.date().millis()})</code></p><p> </p><p><code>    return reduce_region_function</code></p><p> </p><p> </p><p><code>pdsi_function = create_reduce_region_function(scale=5000,</code></p><p><code>                                              reducer=ee.Reducer.mean(),</code></p><p><code>                                              geometry=aoi,</code></p><p><code>                                              crs='EPSG:3310')</code></p><p> </p><p> </p><p><code>pdsi_stat_fc = ee.FeatureCollection(pdsi.map(pdsi_function)).filter(</code></p><p><code>    ee.Filter.notNull(pdsi.first().bandNames()))</code></p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/earthengine-community/pull/152/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
<p>This is much better - thanks for the suggestion! It has been implemented - done.</p>